apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts

```diff
@@ -93,11 +93,11 @@ class MenuPageObject extends BasePage {
   }
 
   get _secondaryComponent() {
-    return By(MENUITEM_TEST_COMPONENT);
+    return By(MENUITEM_TEST_COMPONENT, true);
   }
 
   get _tertiaryComponent() {
-    return By(MENUITEM_NO_A11Y_LABEL_COMPONENT);
+    return By(MENUITEM_NO_A11Y_LABEL_COMPONENT, true);
   }
```

Comment:
@samuelfreiberg Calls to `By` shouldn't need an extra `await` anymore, but in return, users have to specify if they want to start from the root of the element tree. Here, we have to explicitly say to start from the root because we're working with menu items. Is this approach better than the previous?

Reply:
I don't think this is better than previous. I'd rather have two awaits. But still confused - why can't we use: 
`await this._primaryComponent.click();`
instead of 
`await (await this._primaryComponent).click();`

Doesn't WDIO handle this correctly?

Reply:
When you call `$` alone, you don't need that extra await since `$` returns a `ChainablePromiseElement` which allows for `await this._primaryComponent.click()`.

With our queries, we start from an element called `Fluent_Tester_Root_View` which encompasses **most** of the elements in the test app we're searching for. Keyword is **most**. In the case of MenuItems, these spawn outside of `Fluent_Tester_Root_View`. For chained queries, searching for MenuItems returns null, causing an error.

Thus, in our query function, we have to check for this edge case. ```ts
let queryResult: WebdriverIO.Element;
  queryResult = await _rootView.$(selector);
  if (queryResult.error) {
    // In some cases, such as opened ContextualMenu items, the element nodes are not children of the _rootView node, meaning we need to start our search from the top of the tree.
    queryResult = await $(selector);
  }
  return queryResult;
```
Because we have to check if the first query returns null, we have to let the first query resolve into an element. This unfortunately means that we have to work in an async function, and thus can only return a `Promise<Element>` rather than a `ChainablePromiseElement`.